lib/src/gitignore.rs

```diff
@@ -306,37 +211,64 @@ mod tests {
     fn test_gitignore_unusual_symbols() {
         assert!(matches(b"\\*\n", "*"));
         assert!(!matches(b"\\*\n", "foo"));
-        assert!(matches(b"\\\n", "\\"));
         assert!(matches(b"\\!\n", "!"));
         assert!(matches(b"\\?\n", "?"));
         assert!(!matches(b"\\?\n", "x"));
+        assert!(matches(b"\\w\n", "w"));
+        assert!(GitIgnoreFile::empty().chain("", b"\\\n").is_err());
+    }
+
+    #[test]
+    #[cfg(not(target_os = "windows"))]
+    fn teest_gitignore_backslash_path() {
+        assert!(!matches(b"/foo/bar", "/foo\\bar"));
+        assert!(!matches(b"/foo/bar", "/foo/bar\\"));
+
+        assert!(!matches(b"/foo/bar/", "/foo\\bar/"));
+        assert!(!matches(b"/foo/bar/", "/foo\\bar\\/"));
+
         // Invalid escapes are treated like literal backslashes
+        assert!(!matches(b"\\w\n", "\\w"));
+        assert!(matches(b"\\\\ \n", "\\ "));
+        assert!(matches(b"\\\\\\ \n", "\\ "));
+    }
+
+    #[test]
+    #[cfg(target_os = "windows")]
+    /// ignore crate consider backslashes as a directory divider only on
+    /// Windows.
```

Comment:
Do you know if git does the same? Otherwise this seems like a bug that we should report to the `ignore` crate (and maybe help them fix). It doesn't seem important enough to block this PR, though.

Reply:
yes. It looks git also working similarly.
 Mac
```shell
 ~/Downloads/temp ❯ echo /foo/bar > .gitignore
 ~/Downloads/temp ❯ git init .
Initialized empty Git repository in /Users/daehyeok/Downloads/temp/.git/
 ~/Downloads/temp ❯ git check-ignore foo
 ~/Downloads/temp ❯ git check-ignore foo/bar
foo/bar
 ~/Downloads/temp ❯ git check-ignore foo\bar
 ~/Downloads/temp ❯ git check-ignore foo\\bar
 ~/Downloads/temp ❯
```

same test on Windows PowerShell
```shell
PS C:\Users\daehyeok\Downloads\ignore> git check-ignore foo
PS C:\Users\daehyeok\Downloads\ignore> git check-ignore foo/bar
foo/bar
PS C:\Users\daehyeok\Downloads\ignore> git check-ignore foo\bar
"foo\\bar"
PS C:\Users\daehyeok\Downloads\ignore> git ignore foo\\bar
"foo\\\\bar"

```

Reply:
Ah, good. Thanks for checking! I checked the Git spec https://git-scm.com/docs/gitignore before asking and I didn't find anything there. I guess it's an undocumented feature. I'm happy the `ignore` crate matches the behavior.